app/controllers/solidus_stripe/webhooks_controller.rb

```diff
@@ -12,7 +12,8 @@ class WebhooksController < Spree::BaseController
     respond_to :json
 
     def create
-      event = Webhook::Event.from_request(payload: request.body.read, signature_header: signature_header)
+      event = Webhook::Event.from_request(payload: request.body.read, signature_header: signature_header,
```

Comment:
IMHO, that's in the correct layer. The `.from_request` method gets all the request information (the request body, the signature header, and the slug) and transforms it into a webhook event instance. For instance, if it can't find a payment method with that slug, it will return `nil` just like when the payload is malformed, or the signature header is invalid. The controller leverages that to return a bad response code to Stripe. What do you think?

Reply:
I'd prefer not to have yet another method on the payment method model :sweat_smile: (although it's getting thinner), and I'd see it as going way beyond its responsibilities. However, as you suggested, we could pass the request parameter to the `Webhook::Event` class method. That would make the signature simpler, but I think that responsibilities are better defined as they are now:
- The Rails controller knows how to extract things from its `request` method.
- The `Webhook::Event.from_request` would be something more like `Webhook::Event.from_request_information`, less coupled to the Rails internals.
What do you think?

Reply:
That's ok, no need to change the method name, I guess what bothers me (but only a little) is passing the slug, thoughts on letting the controller fetch the payment method from it?

Also probably we can shorten the `signature_header` param to just `signature`:

```rb
Webhook::Event.from_request(
  payload: request.body.read, 
  signature: request.headers[SIGNATURE_HEADER], 
  payment_method: SolidusStripe::WebhookEndpoint.find_by!(slug: params[:slug]).payment_method
)
```

_Anyway consider all of the above as non-blocking nits_ 😊

Reply:
That would only work for some scenarios. If `find_by!` raises, we'd get a 500 response code. Now the query is done from the `.from_request` method, and it rescues the `ActiveRecord::NotFound` error alongside other decoding errors to return a 400 to Stripe.

I'll merge it as it is, but I'm more than happy to keep discussing it!